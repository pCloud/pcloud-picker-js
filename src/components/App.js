// @flow

import * as React from "react";
import pcloudSdk from "pcloud-sdk-js";
import styled from "styled-components";
import { Picker, Modal } from ".";
import { parseItem } from "../utils";
import type { selectedItemType } from "../utils";
import { MODE_SELECT, MODE_UPLOAD } from "../config/constants";

type AppProps = {
  mode: MODE_SELECT | MODE_UPLOAD,
  clientId: string,
  buttonText: string,
  fileUrl: string,
  isFolderSelectionOnly: boolean,
  onSelect: any => void,
  onClose: any => void
};

type AppState = {
  isAuthenticated: boolean,
  isModalOpened: boolean
};

class App extends React.Component<AppProps, AppState> {
  static defaultProps = {
    mode: "",
    clientId: "",
    buttonText: "PcloudButton",
    fileUrl: "",
    isFolderSelectionOnly: false,
    onSelect: () => {},
    onClose: () => {}
  };

  constructor(props: AppProps) {
    super(props);

    this.state = {
      isAuthenticated: false,
      isModalOpened: false
    };

    (this: any).client = null;
    (this: any).receiveToken = this.receiveToken.bind(this);
    (this: any).getToken = this.getToken.bind(this);
    (this: any).getFolderContent = this.getFolderContent.bind(this);
    (this: any).onPick = this.onPick.bind(this);
    (this: any).onCancel = this.onCancel.bind(this);
  }

  openModal() {
    this.setState({ isModalOpened: true });
  }

  closeModal() {
    this.setState({ isModalOpened: false });
  }

  getClient(token: string) {
    // handle errors
    return pcloudSdk.createClient(token);
  }

  receiveToken(token: string) {
    this.client = this.getClient(token);

    this.setState({ isAuthenticated: true, isModalOpened: true });
  }

  getToken() {
    const { clientId } = this.props;
    const { isAuthenticated } = this.state;

    if (!isAuthenticated) {
      pcloudSdk.oauth.initOauthToken({
        client_id: clientId,
        receiveToken: this.receiveToken
      });
    }

    this.openModal();
  }

  getFolderContent(folderId: number) {
    return this.client
      .listfolder(folderId, { iconformat: "id" })
      .then(res => res.contents)
      .then(items => items.map(parseItem))
      .catch(err => {
        console.log(err);
      });
  }

  uploadFile(selectedItemId: string) {
    const { fileUrl } = this.props;

    this.client
      .remoteupload(fileUrl, selectedItemId)
      .then(data => {
        console.log(data);
      })
      .catch(err => console.log(err));
  }

  onPick(selectedItem: selectedItemType) {
    const { mode, onSelect } = this.props;
    const { id } = selectedItem;

    this.closeModal();

    switch (mode) {
      case MODE_SELECT:
        return onSelect(selectedItem);

      case MODE_UPLOAD:
        return this.uploadFile(id);

      default:
        return null;
    }
  }

  onCancel() {
    const { onClose } = this.props;

    onClose();
    this.closeModal();
  }

  render() {
    const { isAuthenticated, isModalOpened } = this.state;
    const { buttonText, isFolderSelectionOnly } = this.props;

    return (
      <Wrapper>
        <Modal show={isModalOpened} closeModal={this.closeModal}>
          {isAuthenticated ? (
            <Picker
              isFolderSelectionOnly={isFolderSelectionOnly}
              getFolderContent={this.getFolderContent}
              onPick={this.onPick}
              onCancel={this.onCancel}
            />
          ) : null}
        </Modal>
        <DefaultButton onClick={this.getToken}>{buttonText}</DefaultButton>
      </Wrapper>
    );
  }
}

export default App;

const Wrapper = styled.div`
  font: 11px/34px Arial, Helvetica;
`;

const DefaultButton = styled.div`
  cursor: pointer;
  background: #20bed6;
  color: #fff;
  margin: 10px;
  padding: 0 20px;
  height: 28px;
  max-width: 150px;
  min-width: 50px;
  width: auto;
  line-height: 28px;
  font-family: Arial, Helvetica;
  font-weight: bold;
  text-align: center;
  border-radius: 3px;
  user-select: none;
`;
