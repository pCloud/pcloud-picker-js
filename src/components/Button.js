import React, { Component } from 'react';
import styled from 'styled-components';

class Button extends Component {

  render() {
    const { text, isDisabled, onButtonClick } = this.props;

    return (
      <DefaultButton
        isDisabled={isDisabled}
        onClick={isDisabled ? null : onButtonClick}
      >
        {text}
      </DefaultButton>
    );
  }
}

export default Button;

const DefaultButton = styled.div`
  cursor: ${props => props.isDisabled ? 'not-allowed' : 'pointer'};
  margin: 10px;
  height: 28px;
  width: 50px;
  text-align: center;
  line-height: 28px;
  background: ${props => props.isDisabled ? '#c1c1c1' : '#20bed6'};
  color: #fff;
  padding: 0 20px;
  border-radius: 3px;
  user-select: none;
`;