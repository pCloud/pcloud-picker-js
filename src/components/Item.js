import React, { Component } from 'react';
import styled from 'styled-components';
import { getIcon } from '../utils';

class Item extends Component {
  constructor() {
    super();

    this._onDoubleClick = this._onDoubleClick.bind(this);
  }

  _onDoubleClick() {
    const { id, name, isFolder, onFolderDoubleClick } = this.props;

    if (isFolder) {
      onFolderDoubleClick(id, name);
    }
  }

  render() {
    const { name, iconId } = this.props;

    return (
      <Row onDoubleClick={this._onDoubleClick}>
        <img src={getIcon(iconId)} alt="icon" />
        <ItemName>{name}</ItemName>
      </Row>
    );
  }
}

export default Item;

const Row = styled.div`
  display: flex;
  align-items: center;
  padding: 0 10px;
  box-sizing: border-box;
  &:hover {
    background-color: #f8f8f8;
  }
`;

const ItemName = styled.div`
  color: #000;
  margin-left: 10px;
  max-width: 429px;
  overflow: hidden;
  text-shadow: 0px 1px 0px rgba(255,255,255,0.5);
  text-overflow: ellipsis;
  user-select: none;
`;