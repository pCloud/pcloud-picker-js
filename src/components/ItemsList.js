// @flow

import * as React from "react";
import { List } from "immutable";
import styled from "styled-components";
import { Item } from ".";

type ItemsListProps = {
  selectedItemId: number,
  items: List<any>,
  onItemClick: () => void,
  onItemDoubleClick: () => void
};

class ItemsList extends React.Component<ItemsListProps, {}> {
  static defaultProps = {
    selectedItemId: "0",
    items: null,
    onItemClick: () => {},
    onItemDoubleClick: () => {}
  };

  constructor(props: ItemsListProps) {
    super(props);

    (this: any)._renderRow = this._renderRow.bind(this);
  }

  _renderRow(item: {
    id: string,
    iconId: number,
    name: string,
    isFolder: boolean
  }) {
    const { selectedItemId, onItemClick, onItemDoubleClick } = this.props;
    const { id, iconId, name, isFolder } = item;

    return (
      <Item
        key={id}
        id={id}
        iconId={iconId}
        name={name}
        isFolder={isFolder}
        isSelected={selectedItemId === id ? true : false}
        onItemClick={onItemClick}
        onItemDoubleClick={onItemDoubleClick}
      />
    );
  }

  _renderItems() {
    const { items } = this.props;

    return items.map(this._renderRow);
  }

  render() {
    const { items } = this.props;

    return (
      <Table>
        {items.size === 0 ? (
          <EmptyFolder>This folder is emptry.</EmptyFolder>
        ) : (
          this._renderItems()
        )}
      </Table>
    );
  }
}

export default ItemsList;

const Table = styled.div`
  width: 100%;
  font-family: "Arial Narrow", Arial;
`;

export const EmptyFolder = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 13px;
  color: #ababab;
`;
