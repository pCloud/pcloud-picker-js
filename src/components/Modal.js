//@ flow

import * as React from "react";
import styled from "styled-components";
import { createPortal } from "react-dom";

type ModalProps = {
  show: boolean,
  children: React.Node,
  onCloseModal: () => void
};

class Modal extends React.Component<ModalProps, {}> {
  renderModal() {
    const { children, show, onCloseModal } = this.props;

    if (!show) {
      return null;
    }

    return [
      <ModalContent>{children}</ModalContent>,
      <ModalOverlay onClick={onCloseModal} />
    ];
  }

  render() {
    const { container } = this.props;

    return createPortal(this.renderModal(), container);
  }
}

export default Modal;

const ModalContent = styled.div`
  position: fixed;
  top: 50%;
  transform: translate(-50%, -50%);
  left: 50%;
  z-index: 5001;
  min-width: 500px;
  width: 30vw;
`;

const ModalOverlay = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  background: grey;
  overflow: hidden;
  opacity: 0.3;
  z-index: 5000;
`;
