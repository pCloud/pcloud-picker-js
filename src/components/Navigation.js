// @flow

import * as React from 'react';
import styled from 'styled-components';
import { List } from 'immutable';
import { NavigationFolder } from '.';

type folder = {
  folderName: string,
  items: null | List<any>
};

type NavigationProps = {
  path: List<number>,
  folders: Map<string, folder>,
  onNameClick: () => void
};

class Navigation extends React.Component<NavigationProps, {}> {
  constructor() {
    super();

    (this: any)._renderFolder = this._renderFolder.bind(this);
  }

  _getFolderName(folderId: string) {
    const { folders } = this.props;

    return folders.getIn([folderId, 'folderName']);
  }

  _renderFolder(folderId: number, index: number) {
    const { onNameClick } = this.props;
    const folderName = this._getFolderName(String(folderId));
    const shouldRenderIcon = index > 0;

    return (
      <NavigationFolder
        key={folderId.toString()}
        id={folderId}
        name={folderName}
        onNameClick={onNameClick}
        shouldRenderIcon={shouldRenderIcon}
      />
    );
  }

  render() {
    const { path } = this.props;

    return (
      //hadndle long path
      <Path>{path.map(this._renderFolder)}</Path>
    );
  }
}

export default Navigation;

const Path = styled.div`
  display: flex;
  padding-left: 10px;
`;