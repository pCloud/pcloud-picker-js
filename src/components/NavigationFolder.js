// @flow

import * as React from "react";
import styled from "styled-components";
import ArrowIcon from "../img/arrow.png";
import { ROOT_FOLDER_ID, ROOT_FOLDER_NAME } from "../config/constants";

type NavigationFolderProps = {
  id: string,
  name: string,
  onNameClick: string => void,
  shouldRenderIcon: boolean
};

class NavigationFolder extends React.Component<NavigationFolderProps, {}> {
  static defaultProps = {
    id: ROOT_FOLDER_ID,
    name: ROOT_FOLDER_NAME,
    onNameClick: () => {},
    shouldRenderIcon: false
  };

  constructor(props: NavigationFolderProps) {
    super(props);

    (this: any)._onClick = this._onClick.bind(this);
  }

  _onClick() {
    const { id, onNameClick } = this.props;

    onNameClick(id);
  }

  render() {
    const { name, shouldRenderIcon } = this.props;

    return (
      <Wrapper>
        {shouldRenderIcon && <img src={ArrowIcon} alt="arrow-icon" />}
        <FolderName onClick={this._onClick}>{name}</FolderName>
      </Wrapper>
    );
  }
}

export default NavigationFolder;

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  height: 30px;
  flex-directions: column;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  word-break: break-all;
`;

const FolderName = styled.div`
  cursor: pointer;
  font-weight: bold;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  word-break: break-all;
  max-width: 100px;
  color: #595959;
`;
