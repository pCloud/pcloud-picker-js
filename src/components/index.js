import Picker from './Picker';
import Navigation from './Navigation';
import NavigationFolder from './NavigationFolder';
import ItemsList from './ItemsList';
import Item from './Item';

export {
  Picker,
  Navigation,
  NavigationFolder,
  ItemsList,
  Item
}