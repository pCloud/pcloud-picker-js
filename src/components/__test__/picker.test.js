import * as React from "react";
import { shallow, mount } from "enzyme";
import "jest-styled-components";

import { Picker, Modal } from "../../components";

describe("<Picker />", () => {
  describe("render", () => {
    it("renders component correctly", () => {
      expect(shallow(<Picker />)).toMatchSnapshot();
    });
  });

  const modalRoot = global.document.createElement("div");
  modalRoot.setAttribute("id", "modal");
  const body = global.document.querySelector("body");
  body.appendChild(modalRoot);

  describe("<DefaultButton />", () => {
    it("should get user token and open modal", () => {
      const picker = mount(<Picker />);
      const defaultButton = picker.find("Picker__DefaultButton");
      expect(modalRoot.hasChildNodes()).toBeFalsy();
      defaultButton.simulate("click");
      expect(modalRoot.hasChildNodes()).toBeDefined();
    });
  });
});
