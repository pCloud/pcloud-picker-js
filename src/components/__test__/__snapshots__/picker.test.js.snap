// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picker /> render renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Picker />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Modal
          container={
            <div
              id="modal"
            />
          }
          onCloseModal={[Function]}
          show={false}
        >
          <Picker__Header>
            <Navigation
              folders={
                Immutable.Map {
                  "0": Object {
                    "folderName": "pCloud",
                    "items": null,
                  },
                }
              }
              onNameClick={[Function]}
              path={
                Immutable.List [
                  "0",
                ]
              }
            />
          </Picker__Header>
          <Picker__Section>
            <Picker__Loader />
          </Picker__Section>
          <Picker__Footer>
            <Picker__DefaultButton
              onClick={[Function]}
            >
              Cancel
            </Picker__DefaultButton>
            <Picker__DefaultButton
              onClick={[Function]}
            >
              Choose
            </Picker__DefaultButton>
          </Picker__Footer>
        </Modal>,
        <Picker__DefaultButton
          onClick={[Function]}
        >
          pCloud
        </Picker__DefaultButton>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Picker__Header>
              <Navigation
                folders={
                  Immutable.Map {
                    "0": Object {
                      "folderName": "pCloud",
                      "items": null,
                    },
                  }
                }
                onNameClick={[Function]}
                path={
                  Immutable.List [
                    "0",
                  ]
                }
              />
            </Picker__Header>,
            <Picker__Section>
              <Picker__Loader />
            </Picker__Section>,
            <Picker__Footer>
              <Picker__DefaultButton
                onClick={[Function]}
              >
                Cancel
              </Picker__DefaultButton>
              <Picker__DefaultButton
                onClick={[Function]}
              >
                Choose
              </Picker__DefaultButton>
            </Picker__Footer>,
          ],
          "container": <div
            id="modal"
          />,
          "onCloseModal": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Navigation
                folders={
                  Immutable.Map {
                    "0": Object {
                      "folderName": "pCloud",
                      "items": null,
                    },
                  }
                }
                onNameClick={[Function]}
                path={
                  Immutable.List [
                    "0",
                  ]
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "folders": Immutable.Map {
                  "0": Object {
                    "folderName": "pCloud",
                    "items": null,
                  },
                },
                "onNameClick": [Function],
                "path": Immutable.List [
                  "0",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Picker__Loader />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Picker__DefaultButton
                  onClick={[Function]}
                >
                  Cancel
                </Picker__DefaultButton>,
                <Picker__DefaultButton
                  onClick={[Function]}
                >
                  Choose
                </Picker__DefaultButton>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Cancel",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cancel",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Choose",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Choose",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "pCloud",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "pCloud",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Modal
            container={
              <div
                id="modal"
              />
            }
            onCloseModal={[Function]}
            show={false}
          >
            <Picker__Header>
              <Navigation
                folders={
                  Immutable.Map {
                    "0": Object {
                      "folderName": "pCloud",
                      "items": null,
                    },
                  }
                }
                onNameClick={[Function]}
                path={
                  Immutable.List [
                    "0",
                  ]
                }
              />
            </Picker__Header>
            <Picker__Section>
              <Picker__Loader />
            </Picker__Section>
            <Picker__Footer>
              <Picker__DefaultButton
                onClick={[Function]}
              >
                Cancel
              </Picker__DefaultButton>
              <Picker__DefaultButton
                onClick={[Function]}
              >
                Choose
              </Picker__DefaultButton>
            </Picker__Footer>
          </Modal>,
          <Picker__DefaultButton
            onClick={[Function]}
          >
            pCloud
          </Picker__DefaultButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Picker__Header>
                <Navigation
                  folders={
                    Immutable.Map {
                      "0": Object {
                        "folderName": "pCloud",
                        "items": null,
                      },
                    }
                  }
                  onNameClick={[Function]}
                  path={
                    Immutable.List [
                      "0",
                    ]
                  }
                />
              </Picker__Header>,
              <Picker__Section>
                <Picker__Loader />
              </Picker__Section>,
              <Picker__Footer>
                <Picker__DefaultButton
                  onClick={[Function]}
                >
                  Cancel
                </Picker__DefaultButton>
                <Picker__DefaultButton
                  onClick={[Function]}
                >
                  Choose
                </Picker__DefaultButton>
              </Picker__Footer>,
            ],
            "container": <div
              id="modal"
            />,
            "onCloseModal": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Navigation
                  folders={
                    Immutable.Map {
                      "0": Object {
                        "folderName": "pCloud",
                        "items": null,
                      },
                    }
                  }
                  onNameClick={[Function]}
                  path={
                    Immutable.List [
                      "0",
                    ]
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "folders": Immutable.Map {
                    "0": Object {
                      "folderName": "pCloud",
                      "items": null,
                    },
                  },
                  "onNameClick": [Function],
                  "path": Immutable.List [
                    "0",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Picker__Loader />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Picker__DefaultButton
                    onClick={[Function]}
                  >
                    Cancel
                  </Picker__DefaultButton>,
                  <Picker__DefaultButton
                    onClick={[Function]}
                  >
                    Choose
                  </Picker__DefaultButton>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Cancel",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Choose",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Choose",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "pCloud",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "pCloud",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
